### PR TITLE
Mock server

### PR DESCRIPTION
**Pull request by miketamis**
First PR ready to go :) LGTM

### CODE REVIEW

**Comment by miketamis** [comment]
Thinking about having the mock data only returned if your querying a new network called "mock" that we make it so that if you query nikau for now it just throws an error. Thoughts? this way we can merge the mock and it makes sense.

**Comment by p-dealwis** [comment]
> Thinking about having the mock data only returned if your querying a new network called "mock" that we make it so that if you query nikau for now it just throws an error. Thoughts? this way we can merge the mock and it makes sense.

I like that idea, that way we can think of having a dedicated mock network for testing purposes. Would make it easier for test automation!

File: schema.graphql [hunk 1]
```diff
@@ -0,0 +1,291 @@
+### This file was generated by Nexus Schema
+### Do not make changes to this file directly
+
+
+type Address implements Node {
+  address: ID
+
+  """
+  Creates a Transcation for signing that will create a new Event that can have tickets.
```
Comment (msgm) [inline schema.graphql:9]: typo Transcation

File: schema.graphql [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+type Query {
+  network(network: NetworkEnum!): Network
+}
+
+"""A ticket for an Ticketed Event"""
+type Ticket {
+  """
+  Creates a Transcation for redeeming the ticket for an entry pass, this manifests as a burn event onchain
+  """
+  createRedeemTranscation: Transcation
```
Comment (msgm) [inline schema.graphql:180]: typo transaction

File: schema.graphql [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+}
+
+type Query {
+  network(network: NetworkEnum!): Network
+}
+
+"""A ticket for an Ticketed Event"""
+type Ticket {
+  """
+  Creates a Transcation for redeeming the ticket for an entry pass, this manifests as a burn event onchain
```
Comment (msgm) [inline schema.graphql:178]: typo transaction

File: schema.graphql [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+  """
+  createRedeemTranscation: Transcation
+
+  """Creates a Transcation for transfering a Ticket to another address"""
```
Comment (msgm) [inline schema.graphql:182]: typo transaction

File: schema.graphql [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+  description: String
+  name: String!
+
+  """The venue for the event"""
+  venue: String
+}
+
+type Transcation {
+  expectedSigningAddress: String
+  transcationData: String
```
Comment (msgm) [inline schema.graphql:285]: typo transaction

File: schema.graphql [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+
+    """Returns the first n elements from the list."""
+    first: Int
+
+    """Returns the last n elements from the list."""
+    last: Int
+  ): TicketConnection
+}
+
+input TicketTypeInput {
```
Comment (msgm) [inline schema.graphql:266]: what is the reasoning behind the 'Input' postfix?

File: schema.graphql [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+  description: String
+  fineprint: String
+  name: String!
+}
+
+input TicketedEventDetailsInput {
```
Comment (msgm) [inline schema.graphql:272]: what is the reasoning behind the 'Input' postfix?

File: schema.graphql [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+  node: CENNZNode
+}
+
+"""
+A date string, such as 2007-12-03, compliant with the `full-date` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
+"""
+scalar Date
+
+type Event {
+  createAdditionalTickets(quantity: Int, ticketTypeId: String): Transcation
```
Comment (msgm) [inline schema.graphql:78]: please describe/comment, typo transaction

File: schema.graphql [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+  """https://facebook.github.io/relay/graphql/connections.htm#sec-Node"""
+  node: CENNZNode
+}
+
+"""
+A date string, such as 2007-12-03, compliant with the `full-date` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
+"""
+scalar Date
+
+type Event {
```
Comment (msgm) [inline schema.graphql:77]: please describe/comment, typo transaction

File: schema.graphql [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+}
+
+"""A momento for attending the event"""
+type TicketStub {
+  """The event"""
+  event: Event
+  name: String
+
+  """The orignal ticket infomation"""
+  orginalTicket: Ticket
```
Comment (msgm) [inline schema.graphql:218]: typo original

File: schema.graphql [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+  """The venue for the event"""
+  venue: String
+}
+
+type Transcation {
+  expectedSigningAddress: String
+  transcationData: String
+}
+
+type TranscationResult {
```
Comment (msgm) [inline schema.graphql:288]: typo transation

File: schema.graphql [hunk 12]
```diff
@@ -0,0 +1,291 @@
...
+    """Returns the last n elements from the list."""
+    last: Int
+  ): CENNZNodeConnection
+}
+
+enum NetworkEnum {
+  Nikau
+}
+
+interface Node {
```
Comment (msgm) [inline schema.graphql:141]: is there a better name we can use? or is this more a blockchain thing?

**Review by msgm — CHANGES_REQUESTED** [review]

File: schema.graphql (hunk 6 above)
Comment (miketamis) [reply schema.graphql:239]: https://graphql.org/graphql-js/mutations-and-input-types/

File: schema.graphql (hunk 12 above)
Comment (miketamis) [reply schema.graphql:114]: https://graphql.org/learn/global-object-identification/#node-interface

File: schema.graphql (hunk 7 above)
Comment (miketamis) [reply schema.graphql:300]: https://graphql.org/graphql-js/mutations-and-input-types/

**Review by msgm — APPROVED** [review]

**Pull request merged**